src/ndarray/ndarray.cc

```diff
@@ -330,11 +330,10 @@ struct NDArrayDLManager {
 };
 
 DLManagedTensor* NDArray::ToDLPack() const {
+  CHECK(!is_none()) << "NDArray is not initialized";
```

Comment:
Just to confirm. Such check should never fail as `ToDLPack` will only be called after `wait_to_read` or `wait_to_write` so the NDArray should always be initialized.

Reply:
Yes. But when users use C++ API directly, they may forget to call WaitToRead or WaitToWrite before calling ToDLPack.